jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -61,17 +90,24 @@ public HttpCookie(String name, String value, String domain, String path, long ma
     }
 
     public HttpCookie(String name, String value, String domain, String path, long maxAge, boolean httpOnly, boolean secure, String comment, int version)
+    {
+        this(name, value, domain, path, maxAge, httpOnly, secure, comment, version, null);
+    }
+
+    public HttpCookie(String name, String value, String domain, String path, long maxAge, boolean httpOnly, boolean secure, String comment, int version, SameSite sameSite)
     {
         _name = name;
         _value = value;
         _domain = domain;
         _path = path;
         _maxAge = maxAge;
-        _httpOnly = httpOnly;
+        // HttpOnly was supported as a comment in cookie flags before the java.net.HttpCookie implementation so need to check that
+        _httpOnly = httpOnly || isHttpOnlyInComment(comment);
         _secure = secure;
-        _comment = comment;
+        _comment = getCommentWithoutFlags(comment);
```

Comment:
I wouldn't call this method from within the constructor.  Specially as this constructor takes both httpOnly and sameSite as parameters, so no need to pass them in as a comment.   The static methods should be there as helpers for a caller that does not have either httpOnly or sameSite